hasktorch/test/Torch/Typed/NN/Recurrent/Cell/GRUSpec.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE DataKinds #-}
+
+module Torch.Typed.NN.Recurrent.Cell.GRUSpec
+  ( Torch.Typed.NN.Recurrent.Cell.GRUSpec.spec
+  )
+where
+
+import           Test.Hspec
+
+import           Torch.HList
+import qualified Torch.NN                      as A
+import qualified Torch.DType                   as D
+import qualified Torch.Device                  as D
+import           Torch.Typed.Factories
+import           Torch.Typed.Functional
+import           Torch.Typed.Parameter
+import           Torch.Typed.NN
+import           Torch.Typed.NN.Recurrent.Cell.GRU
+
+spec :: Spec
+spec = return ()
+
+testGRUCell
+  :: IO
+       (HList
+          '[Parameter '( 'D.CPU, 0) 'D.Float '[21, 5],
+            Parameter '( 'D.CPU, 0) 'D.Float '[21, 7],
+            Parameter '( 'D.CPU, 0) 'D.Float '[21],
+            Parameter '( 'D.CPU, 0) 'D.Float '[21]])
+
+testGRUCell = do
+  let spec = GRUCellSpec  @5 @7 @'D.Float @'( 'D.CPU, 0)
+  model <- A.sample spec
+  pure . flattenParameters $ model
```

Comment:
Thank you for your contribution.
Could you add the test comparing typed spec with value-level's one for gruCell?

Reply:
Yeah sure, will do.

Reply:
Yeah sure, will do.

Reply:
I have added the tests for gruCell as well as lstmCell, can you please take a look?
https://github.com/SurajK7/hasktorch/blob/cdabac8b6be512ada61b47057ccb3d1419ae3316/hasktorch/test/Torch/Typed/FunctionalSpec.hs#L642-L690
https://github.com/SurajK7/hasktorch/blob/cdabac8b6be512ada61b47057ccb3d1419ae3316/hasktorch/test/Torch/Typed/FunctionalSpec.hs#L975-L997

Reply:
Sorry I'm late.
The test is OK. 
Could you push the test for GRUCell branch?